src/test/java/pom/BaseTest.java

```diff
@@ -0,0 +1,146 @@
+                capabilities.setCapability("browserName", "firefox");
+                return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
+            case "grid-safari": // gradle clean test -Dbrowser=grid-safari
+                capabilities.setCapability("browserName", "safari");
+                return driver = new RemoteWebDriver(URI.create(gridURL).toURL(), capabilities);
+            case "lambda": // gradle clean test -Dbrowser=lambda
+                return getLambdaDriver();
+            default:
+                WebDriverManager.chromedriver().setup();
+                ChromeOptions options = new ChromeOptions();
+                options.addArguments("--remote-allow-origins=*");
+                options.addArguments("--disable-notifications");
+                options.addArguments("--start-maximized");
+                // options.addArguments("--headless=new");
+                return driver = new ChromeDriver(options);
+        }
+    }
+    public WebDriver getLambdaDriver() throws MalformedURLException {
+        String userName = "pavlyuk.daria81";
+        String authKey = "fyipWm6hV2KISCPnsQTtgjN7VEyA8QKCkDPAhVP8LI31j5VY8I";
+        String hub = "@hub.lambdatest.com/wd/hub";
+
+        DesiredCapabilities capabilities = new DesiredCapabilities();
+        capabilities.setCapability("platform", "Windows 10");
+        capabilities.setCapability("browserName", "Chrome");
+        capabilities.setCapability("version", "106.0");
+        capabilities.setCapability("resolution", "1024x768");
+        capabilities.setCapability("build", "TestNG With Java");
+        capabilities.setCapability("name", this.getClass().getName());
+        capabilities.setCapability("plugin", "git-testng");
+
+        return new RemoteWebDriver(new URL("https://" + userName + ":" + authKey + hub), capabilities);
+    }
+
+    public void provideEmail(String email) {
+        WebElement emailField = driver.findElement(By.cssSelector("[type ='email']"));
+        emailField.click();
+        emailField.sendKeys(email);
+        wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("[type ='email']")));
+    }
+
+    //step3
+    public void providePassword(String password) {
+        WebElement passwordField = driver.findElement(By.cssSelector("[type='password']"));
+        passwordField.click();
+        passwordField.sendKeys(password);
+        wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("[type='password']")));
+    }
+
+    //step4
+    public void clickSubmit() {
+        WebElement submit = driver.findElement(By.cssSelector("[type ='submit']"));
+        wait.until(ExpectedConditions.presenceOfElementLocated(By.cssSelector("[type ='submit']")));
+        submit.click();
+    }
+
+    public String getPlaylistName() {
+        WebElement playlistElement = driver.findElement(By.cssSelector(".playlist:nth-child(3)"));
+        return playlistElement.getText();
+    }
```

Comment:
**Page Object Model (POM):** recommended to associate the methods with the corresponding page class where they are relevant. In the POM, each page or section of a web application is represented by a separate class. These classes encapsulate the elements and actions specific to that page or section.

**Move these methods to the corresponding class according to the POM.**